packages/@aws-cdk/aws-lambda/lib/lambda-version.ts

```diff
@@ -253,7 +253,7 @@ export class Version extends QualifiedFunctionBase implements IVersion {
     return Lazy.stringValue({
       produce: () => {
         // Validate that the underlying function can be used for Lambda@Edge
-        if (this.lambda instanceof FunctionBase) {
+        if (this.lambda instanceof Function) {
```

Comment:
Curious why you've removed the default implementation of `_checkEdgeCompatibility()` and changed this line.

Why not run `_checkEdgeCompatibility()` if it is any kind of `FunctionBase`?

Reply:
The only reason we needed to implement it at the `FunctionBase` level was because the singleton function's `ensureLambda()` was returning an `IFunction` instead of a `Function` which is incorrect.

Now that the singleton function is fixed we don't need this at the `FunctionBase` level: we will never check imported functions for compatibility.

Reply:
Actually the fact that the type `Function` was aliased as `LambdaFunction` in the `singleton-function.ts` file made me use `FunctionBase` in #9562 because when I tried with `Function` it was in fact referring the a normal function...